frontend/services/gui/GUI_Service.cpp

```diff
@@ -247,6 +243,17 @@ void GUI_Service::digestChangedRequestedResources() {
     /// Get window manipulation resource = resource index 11
     auto& window_manulation = this->m_requestedResourceReferences[11].getResource<megamol::frontend_resources::WindowManipulation>();
     window_manulation.set_mouse_cursor(gui->GetMouseCursor());
+
+    // Register GUI windows only once
+    if (this->register_gui_callbacks_once) {
+
+        auto &screenshot_gui_render_request = this->m_requestedResourceReferences[12].getResource<megamol::frontend_resources::GUIRenderRequest>();
+        gui->RegisterGUIWindow(screenshot_gui_render_request.window_name, screenshot_gui_render_request.callback);
```

Comment:
Exposing a "Add GUI window callback" resource would also relieve the GUI Service from holding possible incoming render requests. Or the other way around: somebody who wants to register his GUI render request/callback does not need to change the GUI Service hpp/cpp to register his callback, if there is a resource that he can request and access in his own service.

Reply:
see commit b3fe3c4